### PR TITLE
match pkce implementation to specifications

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes https://github.com/mozilla/fxa-oauth-server/issues/495

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 8a77576484b076f0a21140907c73b0b499ae2a40
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-oauth-server

---
This message was auto-generated by https://gitcop.com

**Comment by vladikoff** [comment]
This looks good to me @linuxwolf ! 

@rfk final r?

File: lib/validators.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ exports.clientSecret = Joi.string()
   .required();
 
 exports.codeVerifier = Joi.string()
-  .length(32); // https://tools.ietf.org/html/rfc7636#section-4.1
+  .regex(/^[A-Za-z0-9-_]{43,128}$/);  // https://tools.ietf.org/html/rfc7636#section-4.1
```
Comment (rfk) [inline lib/validators.js]: Two tiny nits here:

* Code clarity might be improved by making this regex a shared constant in `validators.B64URL_STRING`
* We might get better error messages if we use `string().min(43).max(128).regex(validators.B64URL_STRING)`

But neither are blockers for me.

Comment (linuxwolf) [reply lib/validators.js]: I'm happy to make those changesmy tomorrow morning (US time).

Comment (vladikoff) [reply lib/validators.js]: Yeah this is a pretty good suggestion, I will update it !

**Comment by vladikoff** [comment]
Continued in https://github.com/mozilla/fxa-oauth-server/pull/498

**Pull request closed without merging**